### DISCUSSION TITLE
pip-compile with depending requirements files

### DISCUSSION THREAD

**Discussion by ods** [renovatebot/renovate#25732] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.57.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

It's commonly used pattern to split requirements into 2 separate files, one for application itself (`requirements/base.in` in my case, used to generate `requirements/base.txt`) and one for development (`requirements/dev.in`, used to generate `requirements/dev.txt`). To keep dependencies in agreement, the first line in `requirements/dev.in` is a constraint option:
```
-c base.txt
# ... dev packages
```

For this pattern implies some restrictions. Namely, `requirements/dev.txt` can be updated independently (using current `requirements/base.txt` as constraint), while `requirements/base.txt` cannot be: after  updating it you must also update `requirements/dev.txt` to keep versions in sync. Thus, the order of updates is vital. 

And this is what doesn't work: renovate always update dev dependencies first, using old base requirements file. So, when base dependencies are updated, they are no longer in agreement. This doesn't work even when I explicitly list them in required order in config file:
```json5
    "pip-compile": {
        "fileMatch": [
            // The order is important
            "^requirements/base\\.in$",
            "^requirements/dev\\.in$",
        ],
    }
```

Another (rather minor) problem is that renovate uses wrong working directory when runs command. This undesirable changes to the diff:
```diff
- #    pip-compile --generate-hashes --strip-extras requirements/base.in
+ #    pip-compile --generate-hashes --strip-extras base.in
```

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by not7cd** [comment]
This will be resolved by supporting relative and absolute paths

**Reply by not7cd** [reply]
@ods, this feature should be available in the recent release. It was released here in [37.203.0](https://github.com/renovatebot/renovate/releases/tag/37.203.0). But there were multiple fixes in the meantime, so use newest. I will be grateful if you can confirm it working.
